tests/test_cargo_rustc.rs

```diff
@@ -0,0 +1,293 @@
+use std::path::MAIN_SEPARATOR as SEP;
```

Comment:
Actually, what I would have liked is the ability to check the cargo ouput for `not matches`, e.g. modify the `support::lines_match` to also accept a `[!..]` wildcard, or something similar. By doing this we could check for `[..] <opts>... [..]` occurrences where we want the args to through, and make sure that the other compilations does not contain the extra args like `[!..] <opts>... [!..]`.